app/controllers/concerns/europeana/logging/log_session_id.rb

```diff
@@ -8,15 +8,19 @@ module LogSessionId
       # Overrides `#logger` in controller to first set session via
       # `Europeana::Logging::SessionLogging#session=`
       def logger(*args)
-        super.session_id = session.id unless @_request.nil?
+        super.session_id = session_id_from_session unless @_request.nil?
         super(*args)
       end
 
       ##
       # Lograge payload
       def append_info_to_payload(payload)
         super
-        payload[:session_id] = session.id
+        payload[:session_id] = session_id_from_session unless session_id_from_session.blank?
+      end
+
+      def session_id_from_session
+        session.is_a?(Hash) ? session[:id] : session.id
```

Comment:
On reflection, is `session[:id]` ever going to contain anything?

We know `session` is a Hash when the controller inherits from `ActionController::API`, but then session id will always be empty.

So should we rather do this?
```ruby
session.respond_to?(:id) ? session.id : nil
```

Reply:
yes, that makes more sense for now. Should we ever have alternative ways to retrieve a session ID in the future we should implement them then and not try to anticipate anything like this.